src/js/src/controllers/issue-tracker.controller.js

```diff
@@ -1,266 +1,251 @@
-                    vm.issueList.push(issueList);
-                    vm.initIssue = vm.copyIssue();
-                    vm.sendIssueData();
-                }
-            });
-        };
-
-        vm.editIssue = function (issue, index) {
-            var modalInstance = issueEditService.create(issue, vm.priorities, vm.stasuses, index);
-
-            modalInstance.result.then(function (issueList) {
-
-                if (issueList !== null) {
-                    _.extend(issue, issueList);
-                    vm.initIssue = vm.copyIssue();
-                    vm.sendIssueData();
-                } else {
-                    var modalInstance = dialogService.create('d');
-
-                    modalInstance.result.then(function (result) {
-
-                        if (result !== null) {
-                            vm.issueList.splice(index, 1);
-                            vm.sendIssueData();
-                        }
-                    });
-                }
-                vm.sendIssueData();
-            });
-        };
-
-        vm.deleteIssue = function (index) {
-
-            var modalInstance = dialogService.create('d');
-
-            modalInstance.result.then(function (result) {
-
-                if (result !== null) {
-                    vm.issueList.splice(index, 1);
-                    vm.sendIssueData();
-                }
-            });
-        };
-
-        vm.help = function () {
-            var modalInstance = helpService.create(vm.poll, vm.count, vm.log);
-
-            modalInstance.result.then(function (count) {
-                if (count !== null) {
-                    vm.count = count;
-                    vm.sendPoll();
-                }
-            });
-        };
-
-        vm.settings = function () {
-            var modalInstance = settingsService.create(vm.colorScheme, vm.stasuses, vm.scheme);
-
-            modalInstance.result.then(function (scheme) {
-                if (scheme !== null) {
-                    vm.scheme = scheme;
+        });
+    };
+
+    getHelp() {
+        let pollObject = {
+            poll: this.poll,
+            count: this.count,
+            log: this.log
+        };
+        this.modalService.getHelpModal(pollObject).result.then();
+    };
+
+    getDialog(issue, index) {
+        this.updIssueList[index] = this.issueList[index];
+        this.issueList[index] = this.initIssue[index];
+        this.modalService.getDialogModal('c').result.then((result) => {
+            if (result) {
+                this.issueList[index] = this.updIssueList[index];
+                this.sendIssueData();
+            }
+            else if (!result) {
+                this.issueList = this.copyIssue()
+            }
+        });
+    };
+
+    copyIssue() {
+        return angular.copy(this.issueList)
+    };
+
+    getRemainingIssues() {
+        let count;
+        count = _.filter(this.issueList, (issue) => {return issue.status.id !== 3}).length;
+        return count;
```

Comment:
`return _.filter(this.issueList, (issue) => {return issue.status.id !== 3}).length;`